x-pack/plugins/apm/public/components/app/Settings/agent_keys/create_agent_key.tsx

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useState, useEffect } from 'react';
+import { i18n } from '@kbn/i18n';
+import {
+  EuiButton,
+  EuiFlyout,
+  EuiFlyoutHeader,
+  EuiTitle,
+  EuiFlyoutBody,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFlyoutFooter,
+  EuiButtonEmpty,
+  EuiForm,
+  EuiFormRow,
+  EuiSpacer,
+  EuiFieldText,
+  EuiText,
+  EuiFormFieldset,
+  EuiCheckbox,
+  htmlIdGenerator,
+} from '@elastic/eui';
+import { isEmpty } from 'lodash';
+import { callApmApi } from '../../../../services/rest/createCallApmApi';
+import { useKibana } from '../../../../../../../../src/plugins/kibana_react/public';
+import { ApmPluginStartDeps } from '../../../../plugin';
+import { CreateApiKeyResponse } from '../../../../../common/agent_key_types';
+
+interface Props {
+  onCancel(): void;
+  onSuccess: (agentKey: CreateApiKeyResponse) => void;
+  onError: (keyName: string) => void;
+}
+
+export function CreateAgentKeyFlyout({ onCancel, onSuccess, onError }: Props) {
+  const {
+    services: { security },
+  } = useKibana<ApmPluginStartDeps>();
+
+  const [username, setUsername] = useState('');
+
+  const [formTouched, setFormTouched] = useState(false);
+  const [keyName, setKeyName] = useState('');
+  const [agentConfigChecked, setAgentConfigChecked] = useState(true);
+  const [eventWriteChecked, setEventWriteChecked] = useState(true);
+  const [sourcemapChecked, setSourcemapChecked] = useState(true);
+
+  const isFormInvalid = isEmpty(keyName);
+  const formError = i18n.translate(
+    'xpack.apm.settings.agentKeys.createKeyFlyout.name.placeholder',
+    {
+      defaultMessage: 'Enter a name',
+    }
+  );
+
+  useEffect(() => {
+    const getCurrentUser = async () => {
+      try {
+        const authenticatedUser = await security?.authc.getCurrentUser();
```

Comment:
It would be better to avoid making another request to get the authenticated user but other places are doing this as well, including api keys view in the security plugin.